cli/src/core/parse/packageSpec.go

```diff
@@ -0,0 +1,79 @@
+package parse
+
+import (
+	"cli/core"
+	"cli/core/state"
+
+	"github.com/luno/jettison/errors"
+	"github.com/spf13/cobra"
+)
+
+func getCustomPackages(config *core.Config, customPackagePaths []string) []core.CustomPackage {
+	var customPackages []core.CustomPackage
+	for _, customPackagePath := range customPackagePaths {
+		var customPackage core.CustomPackage
+
+		for _, configCustomPackage := range config.CustomPackages {
+			if customPackagePath == configCustomPackage.Id || customPackagePath == configCustomPackage.Path {
+				customPackage = configCustomPackage
+				break
+			}
+		}
+		if customPackage.Id == "" {
+			customPackage = core.CustomPackage{
+				Path: customPackagePath,
+			}
+		}
+
+		customPackages = append(customPackages, customPackage)
+	}
+	return customPackages
+}
+
+func GetPackageSpecFromParams(cmd *cobra.Command, config *core.Config) (*core.PackageSpec, error) {
+	packageSpec := core.PackageSpec{}
+
+	packageNames, err := cmd.Flags().GetStringSlice("name")
+	if err != nil {
+		return nil, errors.Wrap(err, "")
+	}
+	customPackagePaths, err := cmd.Flags().GetStringSlice("custom-path")
+	if err != nil {
+		return nil, errors.Wrap(err, "")
+	}
+	isDev, err := cmd.Flags().GetBool("dev")
+	if err != nil {
+		return nil, errors.Wrap(err, "")
+	}
+	isOnly, err := cmd.Flags().GetBool("only")
+	if err != nil {
+		return nil, errors.Wrap(err, "")
+	}
+
+	var envVariables []string
+	if cmd.Flags().Changed("env-file") {
+		envFiles, err := cmd.Flags().GetStringSlice("env-file")
+		if err != nil {
+			return nil, errors.Wrap(err, "")
+		}
+
+		envViper, err := state.GetEnvironmentVariableViper(envFiles)
+		if err != nil {
+			return nil, err
+		}
+		envVariables = state.GetEnvVariableString(envViper)
+	}
+
+	customPackages := getCustomPackages(config, customPackagePaths)
+
+	packageSpec = core.PackageSpec{
+		Packages:             packageNames,
+		CustomPackages:       customPackages,
+		EnvironmentVariables: envVariables,
+		IsDev:                isDev,
+		IsOnly:               isOnly,
+		DeployCommand:        cmd.Use,
+	}
```

Comment:
The `PackageSpec` type has the `ImageVersion` property. should we not specify it here?

Reply:
It isn't here because currently the user doesn't have the option to specify the image version through the command line, if you want this as a feature, I can add a TODO, then make a new PR for that.